src/lib.rs

```diff
@@ -224,145 +237,259 @@ pub fn set_symlink_file_times<P>(p: P, atime: FileTime, mtime: FileTime)
     set_symlink_file_times_(p.as_ref(), atime, mtime)
 }
 
-#[cfg(unix)]
-fn set_file_times_(p: &Path, atime: FileTime, mtime: FileTime) -> io::Result<()> {
-    use libc::utimes;
-    fn set_time(filename: *const c_char, times: *const timeval) -> c_int {
-        unsafe {
-            utimes(filename, times)
+// More generally available, but provides only ms-grain precision.
+#[cfg(any(target_os = "macos",
```

Comment:
I think these `#[cfg]` annotations could avoid duplication if the module is always called "imp" (or some similar name), right?

Reply:
oh I guess that is the pattern we use, isn't it. :)

Reply:
(will fix after the travis runs complete so that I can first learn about other potential latent bugs in the existing code.)